app/views/users/_status.html.erb

```diff
@@ -13,11 +14,27 @@
         <tr>
           <td><%= lesson.language.name %></td>
           <td><%= lesson.teacher.first_name %> <%= lesson.teacher.last_name %></td>
-          <td><%= lesson.date %></td>
-          <% if lesson.status == "completed" %>
-            <td><%= link_to "Add review", users_dashboard_path, class: "btn btn-ghost" %></td>
+          <td><%= lesson.date.strftime("%b %e, %l:%M hs") %></td>
+          <td><%= lesson.location %></td>
+
+          <% case state %>
+          <% when "confirmed" %>
+            <td><%= link_to "Cancel", lesson_path(lesson, lesson: { status: "canceled" }), method: :patch, class: "btn btn-ghost" %></td>
+          <% when "pending" %>
+            <% if @user.is_teacher %>
+              <td><%= link_to "Confirm", lesson_path(lesson, lesson: { status: "confirmed" }), method: :patch, class: "btn btn-success" %></td>
+              <td><%= link_to "Cancel", lesson_path(lesson, lesson: { status: "canceled" }), method: :patch, class: "btn btn-ghost" %></td>
+            <% else %>
+              <td><%= link_to "Cancel", lesson_path(lesson, lesson: { status: "canceled" }), method: :patch, class: "btn btn-ghost" %></td>
+            <% end %>
+          <% when "completed" %>
+            <% if @user.is_teacher %>
+              <td><%= link_to "Delete", lesson_path(lesson), method: :delete, class: "btn btn-ghost " %></td>
+            <% else %>
+              <td><%= link_to "Add review", new_lesson_review_path(lesson), class: "btn btn-ghost" %></td>
+            <% end %>
           <% else %>
-            <td><%= link_to "Contact teacher", root_path, class: "btn btn-ghost" %></td>
+              <td><%= link_to "Delete", lesson_path(lesson), method: :delete, class: "btn btn-ghost" %></td>
```

Comment:
No delete option - one day, we could theoretically add an ARCHIVE status too, but for now, I think we should remove the delete option